integration/local-gadget/integration_test.go

```diff
@@ -0,0 +1,256 @@
+	"flag"
+	"fmt"
+	"os"
+	"reflect"
+	"strings"
+	"testing"
+
+	. "github.com/kinvolk/inspektor-gadget/integration"
+	containercollection "github.com/kinvolk/inspektor-gadget/pkg/container-collection"
+	bindTypes "github.com/kinvolk/inspektor-gadget/pkg/gadgets/trace/bind/types"
+)
+
+const (
+	ContainerRuntimeDocker     = "docker"
+	ContainerRuntimeContainerd = "containerd"
+	ContainerRuntimeCRIO       = "cri-o"
+)
+
+var supportedContainerRuntimes = []string{ContainerRuntimeDocker, ContainerRuntimeContainerd, ContainerRuntimeCRIO}
+
+var (
+	integration      = flag.Bool("integration", false, "run integration tests")
+	containerRuntime = flag.String("container-runtime", "", "allows to do validation for expected runtime in the tests")
+)
+
+func TestMain(m *testing.M) {
+	flag.Parse()
+	if !*integration {
+		fmt.Println("Skipping integration test.")
+		os.Exit(0)
+	}
+
+	if *containerRuntime != "" {
+		found := false
+		for _, val := range supportedContainerRuntimes {
+			if *containerRuntime == val {
+				found = true
+				break
+			}
+		}
+
+		if !found {
+			fmt.Fprintf(os.Stderr, "Error: invalid argument '-container-runtime': %q. Valid values: %s\n",
+				*containerRuntime, strings.Join(supportedContainerRuntimes, ", "))
+			os.Exit(-1)
+		}
+	} else {
+		fmt.Println("Error: '-container-runtime' must be specified")
+		os.Exit(1)
+	}
+
+	fmt.Println("Start running tests:")
+	os.Exit(m.Run())
+}
+
+func TestListContainers(t *testing.T) {
+	t.Parallel()
+	ns := GenerateTestNamespaceName("test-list-containers")
+
+	listContainersCmd := &Command{
+		Name: "RunListContainers",
+		Cmd:  fmt.Sprintf("local-gadget list-containers -o json --runtimes=%s", *containerRuntime),
+		ExpectedOutputFn: func(output string) error {
+			expectedContainer := &containercollection.Container{
+				Podname:   "test-pod",
+				Runtime:   *containerRuntime,
+				Namespace: ns,
+			}
+
+			normalize := func(c *containercollection.Container) {
+				c.Name = ""
+				c.ID = ""
+				c.Pid = 0
+				c.OciConfig = nil
+				c.Bundle = ""
+				c.Mntns = 0
+				c.Netns = 0
+				c.CgroupPath = ""
+				c.CgroupID = 0
+				c.CgroupV1 = ""
+				c.CgroupV2 = ""
+				c.Labels = nil
+				c.PodUID = ""
+			}
+
+			var containers []*containercollection.Container
+			if err := json.Unmarshal([]byte(output), &containers); err != nil {
+				return err
+			}
+
+			for _, gotContainer := range containers {
+				normalize(gotContainer)
+				if reflect.DeepEqual(gotContainer, expectedContainer) {
+					return nil
+				}
+			}
+
+			return fmt.Errorf("output doesn't contain the expected container: %+v", expectedContainer)
+		},
+	}
+
+	commands := []*Command{
+		CreateTestNamespaceCommand(ns),
+		BusyboxPodRepeatCommand(ns, "sleep inf"),
+		WaitUntilTestPodReadyCommand(ns),
+		listContainersCmd,
+		DeleteTestNamespaceCommand(ns),
+	}
+
+	RunCommands(commands, t)
+}
+
+func TestListSingleContainer(t *testing.T) {
+	t.Parallel()
+	prefix := "test-list-single"
+	po := fmt.Sprintf("%s-pod", prefix)
```

Comment:
nit: is `po` and easy to understand variable name?

Reply:
I saw that we are using `ns` so I thought using short form from `kubectl` for pods i.e `po` would be okay. What do you say?

Reply:
I didn't know `po` was common. Let's keep po, that should be fine.